build-tools/packages/build-tools/src/fluidBuild/tasks/taskUtils.ts

```diff
@@ -97,3 +118,75 @@ export async function loadModule(modulePath: string, moduleType?: string) {
 	}
 	return require(modulePath);
 }
+
+/**
+ * Shuffles an array in place using Fisher-Yates algorithm.
+ * Used for testing order-independence when FLUID_BUILD_TEST_RANDOM_ORDER is set.
+ *
+ * @param array - The array to shuffle
+ * @returns The shuffled array (same reference, modified in place)
+ */
+function shuffleArray<T>(array: T[]): T[] {
+	for (let i = array.length - 1; i > 0; i--) {
+		const j = Math.floor(Math.random() * (i + 1));
+		[array[i], array[j]] = [array[j], array[i]];
+	}
+	return array;
+}
```

Comment:
Super nit: since it shuffles in place, I think an API that doesn't return anything might prevent bugs where people assume the original array remains untouched because they got a new one.